src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -47,91 +51,120 @@ local campaignTokenData = {
-    broadcastToAll("Could not find importer token", Color.Red)
-  elseif #coinObjects > 1 then
-    broadcastToAll("More than 1 importer token found. Please delete all but 1 importer token", Color.Yellow)
-  else
-    coin = coinObjects[1]
+function onCollisionEnter(info)
+  local objectData = info.collision_object.getData()
+  if info.collision_object.hasTag("ImporterToken") then
+    importFromToken(info.collision_object)
+  end
+end
 
-    local importData = JSON.decode(coin.getGMNotes())
-    campaignBoxGUID = importData["box"]
+-- Identifies import token, determines campaign box and downloads it (if needed)
+function importFromToken(coin)
+  broadcastToAll("Campaign Import Initiated")
+  local importData = JSON.decode(coin.getGMNotes())
 
-    local campaignBox = getObjectFromGUID(campaignBoxGUID)
-    if campaignBox.type == "Generic" then
-      campaignBox.call("buttonClick_download")
-    end
-    Wait.condition(
-      function()
-        if #campaignBox.getObjects() > 0 then
-          placeCampaignFromToken(importData)
-        else
-          createCampaignFromToken(importData)
-        end
-      end,
-      function()
-        local obj = getObjectFromGUID(campaignBoxGUID)
-        if obj == nil then
-          return false
-        else
-          return obj.type == "Bag" and obj.getLuaScript() ~= ""
-        end
-      end,
-      2,
-      function() broadcastToAll("Error loading campaign box") end
-    )
+  local campaignBoxGUID = importData["box"]
+  local campaignBox = getObjectFromGUID(campaignBoxGUID)
+  if not campaignBox then
+    broadcastToAll("Campaign Box not present on table!", Color.Red)
+    return
   end
+  if campaignBox.type == "Generic" then
+    campaignBox.call("buttonClick_download")
+  end
+  Wait.condition(
+    function()
+      if #campaignBox.getObjects() > 0 then
+        placeCampaignFromToken(importData, coin)
+      else
+        createCampaignFromToken(importData, coin)
+      end
+    end,
+    function()
+      local obj = getObjectFromGUID(campaignBoxGUID)
+      if obj == nil then
+        return false
+      else
+        return obj.type == "Bag" and obj.getLuaScript() ~= ""
+      end
+    end,
+    2,
+    function() broadcastToAll("Error loading campaign box") end
+  )
 end
 
 -- After box has been downloaded, places content on table
-function placeCampaignFromToken(importData)
+function placeCampaignFromToken(importData, coin)
   getObjectFromGUID(importData["box"]).call("buttonClick_place")
   Wait.condition(
-    function() createCampaignFromToken(importData) end,
+    function() createCampaignFromToken(importData, coin) end,
     function() return findUniqueObjectWithTag("CampaignLog") ~= nil end,
     2,
     function() broadcastToAll("Error placing campaign box") end
   )
 end
 
 -- After content is placed on table, conducts all the other import operations
-function createCampaignFromToken(importData)
+function createCampaignFromToken(importData, coin)
+
+  checkWarning = false
```

Comment:
Not a big fan of the frequent enabling / disabling of this

Reply:
It's not really frequent though?  It only happens when importing or exporting

Reply:
Yeah, it's at multiple points though. Why not just disable it at the start and keep it disabled for the whole process instead of toggling it?

Reply:
??? It's only disabled once (on this line), then reenabled shortly thereafter.  Don't see the issue?

Reply:
Okay I must have been looking at an old version when I made that comment? Or I was just sleepy.. Either way this works nicely